php/src/Operation.cpp

```diff
@@ -86,35 +84,34 @@ class OperationI : public Operation
     static ParamInfoPtr convertParam(zval*, int);
     static void getArgInfo(zend_internal_arg_info&, const ParamInfoPtr&, bool);
 };
-typedef IceUtil::Handle<OperationI> OperationIPtr;
+using OperationIPtr = std::shared_ptr<OperationI>;
 
-//
 // The base class for client-side invocations.
-//
-class Invocation : public IceUtil::Shared
+class Invocation
 {
 public:
 
-    Invocation(const Ice::ObjectPrx&, const CommunicatorInfoPtr&);
+    virtual ~Invocation() = default;
+
+    Invocation(shared_ptr<Ice::ObjectPrx>, CommunicatorInfoPtr);
```

Comment:
I assume you removed the const& because the ctor adopts its parameters?
Make them `&&` would be clearer.

Reply:
I thought it was best to pass by value, as we do in the new C++ mapping

Reply:
The situation is different for mapped server-side operations in the new C++ mapping.

For these functions, the consumer is the developer implementing the function, and it's nicer to give the choice to just use the parameter (very common) or move it into something else for further transmission / storage.

Reply:
And alternatively, if the call site typically does not have a rvalue to give, then keep these parameters const&.